library/src/main/cpp/util.cpp

```diff
@@ -17,26 +18,35 @@ namespace util {
 
     std::mutex util_mutex_ = std::mutex();
 
-    jbyteArray bytes_from_ustring(JNIEnv* env, session::ustring_view from_str) {
-        size_t length = from_str.length();
+    jbyteArray bytes_from_vector(JNIEnv* env, std::vector<unsigned char> from_str) {
```

Comment:
So previously it takes in the `ustring_view` which is just a fancy pointer + len but `std::vector` will make a new copy of the bytes?

So potentially we take in a reference instead? Since it copies the data to Java land anyway.
```suggestion
    jbyteArray bytes_from_vector(JNIEnv* env, const std::vector<unsigned char> &from_str) {
```

Reply:
Ah yep good catch